static/js/compose_actions.js

```diff
@@ -244,6 +244,13 @@ export function start(msg_type, opts) {
         opts.private_message_recipient = "";
     }
 
+    if (
+        compose_state.composing() &&
+        (opts.trigger === "popover send private" || opts.trigger === "user sidebar popover")
```

Comment:
Why just these two triggers?  I would have expected a conditional more alone the lines of "is the compose box empty".

Reply:
I was thinking along the issue of updating drafts just for switching to PM popovers (which is inefficient).
But, I've updated the code to use the `has_message_content()` function which takes care of updating drafts whenever we have some content and we're about to clear the compose box.

Reply:
`update_draft` itself already does that check, via `snapshot_message`.